### PR TITLE
.NET Core transition: Removed (almost) all instances of Current.ShortStringHlper and replaced with constructor injection (#7371)

### PR DESCRIPTION
**Pull request by AndyButland**
This PR is for issue #7371 ".NET Core transition: Inject IShortStringHelper wherever possible" where I've completed the task detailed in the issue - to replace use of `Current.ShortStringHelper` with constructor injected versions - with two exceptions:

- `MemberSaveValidationAttribute`
- `MemberBinder`

At a first look, it seems these two classes need to have parameterless constructors, so I'm not sure how to approach removing the dependency on  `Current.ShortStringHelper` here.  I've amended the alternate constructor.  Maybe a reviewer has some ideas how to remove these last two usages in the code base..

The platform builds and unit tests pass (though I did have issues running the tests in `Umbraco.Tests.Services.ContentTypeServiceTests` and `Umbraco.Tests.Services.ContentTypeServiceVariantsTests`, so they should be checked.  I was getting "out of memory" errors (not sure if that's a sign the test is doing a bit much, or if I need a new laptop!).  But I was seeing these on the main `netcore/dev` branch too.

Umbraco also boots and runs, and a cursory click around the back-office didn't trigger any errors that I could see.

### CODE REVIEW

**Comment by Shazwazza** [comment]
Great work ! I'll add a note to our sprint to get thsi reviewed as soon as we can. for attributes, since they are static, we have no choice but to use service locators, but would be good to have ctor overloads for DI for testing.

**Comment by Shazwazza** [comment]
Hi @AndyButland with recent changes there's now some merge conflicts. If you have time can you fix those up?

**Comment by AndyButland** [comment]
Sure, have updated.

File: src/Umbraco.Web/Editors/MacrosController.cs [hunk 1]
```diff
@@ -35,7 +34,7 @@ public class MacrosController : BackOfficeNotificationsController
         private readonly IMacroService _macroService;
 
         public MacrosController(IGlobalSettings globalSettings, IUmbracoContextAccessor umbracoContextAccessor, ISqlContext sqlContext, ServiceContext services, AppCaches appCaches, IProfilingLogger logger, IRuntimeState runtimeState, UmbracoHelper umbracoHelper, IShortStringHelper shortStringHelper)
-            : base(globalSettings, umbracoContextAccessor, sqlContext, services, appCaches, logger, runtimeState, umbracoHelper)
+            : base(globalSettings, umbracoContextAccessor, sqlContext, services, appCaches, logger, runtimeState, umbracoHelper, shortStringHelper)
         {
             _shortStringHelper = shortStringHelper;
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Web/Editors/MacrosController.cs]: Agreed.

File: src/Umbraco.Web/Editors/UmbracoAuthorizedJsonController.cs [hunk 2]
```diff
@@ -24,9 +26,12 @@ protected UmbracoAuthorizedJsonController()
...
         }
 
-        protected UmbracoAuthorizedJsonController(IGlobalSettings globalSettings, IUmbracoContextAccessor umbracoContextAccessor, ISqlContext sqlContext, ServiceContext services, AppCaches appCaches, IProfilingLogger logger, IRuntimeState runtimeState, UmbracoHelper umbracoHelper)
+        protected UmbracoAuthorizedJsonController(IGlobalSettings globalSettings, IUmbracoContextAccessor umbracoContextAccessor, ISqlContext sqlContext, ServiceContext services, AppCaches appCaches, IProfilingLogger logger, IRuntimeState runtimeState, UmbracoHelper umbracoHelper, IShortStringHelper shortStringHelper)
             : base(globalSettings, umbracoContextAccessor, sqlContext, services, appCaches, logger, runtimeState, umbracoHelper)
         {
+            ShortStringHelper = shortStringHelper ?? throw new ArgumentNullException(nameof(shortStringHelper));
         }
+
+        public IShortStringHelper ShortStringHelper { get; }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Web/Editors/UmbracoAuthorizedJsonController.cs]: No, I agree it's better made `protected`.

**Pull request merged**